test-data/exceptions.test

```diff
@@ -187,3 +187,82 @@ L12:
 L13:
     dec_ref r10
     goto L7
+
+[case testTryFinally2Asdf]
+def a() -> str:
+    try:
+        print()
+        return 'hi'
+    finally:
+        pass
+[out]
+L0:
+L1:
+    r1 = builtins.module :: static
+    r2 = unicode_0 :: static  ('print')
+    r3 = getattr r1, r2
+    if is_error(r3) goto L7 (error at a:3) else goto L2
+L2:
+    r4 = py_call(r3)
+    dec_ref r3
+    if is_error(r4) goto L7 (error at a:3) else goto L3
+L3:
+    r5 = cast(None, r4)
+    if is_error(r5) goto L7 (error at a:3) else goto L21
+L4:
+    r6 = unicode_1 :: static  ('hi')
+    inc_ref r6
+    r0 = r6
+    goto L6
+L5:
+    r8 = <error> :: str
+    r0 = r8
+L6:
+    r9 = <error> :: tuple[object, object, object]
+    r7 = r9
+    goto L8
+L7:
+    r10 = <error> :: str
+    r0 = r10
+    r11 = err_catch
+    r7 = r11
+L8:
+    if is_error(r7) goto L11 else goto L9
+L9:
+    reraise_exc; r12 = 0
+    if not r12 goto L14 else goto L22 :: bool
+L10:
+    unreachable
```

Comment:
Why is this unreachable? There is a goto that targets this block below (from L22).

Reply:
That will never get reached, though. The problem is that exception raising is kind of hacky: we don't have an ERR_ALWAYS, so things that always through are ERR_FALSE and conditional, even though they really aren't.

Reply:
(Filed #261 re: this)